shard/src/ext/latex.py

```diff
@@ -0,0 +1,51 @@
+# Latex command for Architus bot.
+# Takes a string of latex and returns the compiled version as a png
+# Inspired by: https://github.com/chuanshi/slacklatex
+from discord.ext import commands
+from discord import File
+
+from string import Template
+import tempfile
+import os
+from asyncio import create_subprocess_exec
+from asyncio.subprocess import DEVNULL
+
+
+@commands.command(aliases=['tex'])
+async def latex(ctx, latex):
+    with tempfile.TemporaryDirectory() as work_dir:
+        with open('src/ext/template.tex', 'r',) as f:
+            s = Template(f.read())
+        out_txt = s.substitute(my_text=latex)
+
+        with open(os.path.join(work_dir, 'out.tex'), 'w') as f:
+            f.write(out_txt)
```

Comment:
ideally we would use `bytesIO` for this instead of writing to disk.  I guess I'll merge without that since it'd probably be miserable to implement and this should work

Reply:
The standard latex compiler does not have any capability of not writing to disk from what I could tell reading the man page and a quick google search. The closest option is having it write the output to a socket but even then it still writes the file to disk. Implementing a way that doesn't involve writing to disk would require moving to a completely different latex compiler.